src/ZODB/Connection.py

```diff
@@ -302,8 +302,8 @@ def close(self, primary=True):
                     f()
                 except: # except what?
                     f = getattr(f, 'im_self', f)
-                    self._log.error("Close callback failed for %s", f,
-                                    exc_info=sys.exc_info())
+                    self._log.error("Close callback failed for %s %s", f,
+                                    sys.exc_info())
             self.__onCloseCallbacks = None
```

Comment:
Apologies, I misspoke: your change doesn't remove the exception type it's part of the 3-tuple returned by exc_info()), but still, the change in the formatting and the loss of the traceback are not good.